Objects/typeobject.c

```diff
@@ -8296,17 +8296,22 @@ _Py_slot_tp_getattr_hook(PyObject *self, PyObject *name)
     if (getattribute == NULL ||
         (Py_IS_TYPE(getattribute, &PyWrapperDescr_Type) &&
          ((PyWrapperDescrObject *)getattribute)->d_wrapped ==
-         (void *)PyObject_GenericGetAttr))
-        res = PyObject_GenericGetAttr(self, name);
-    else {
+             (void *)PyObject_GenericGetAttr)) {
+        res = _PyObject_GenericGetAttrWithDict(self, name, NULL, 1);
+        /* if no error has occurred, then it must be suppressed by us */
+        if (res == NULL && !PyErr_Occurred()) {
+            res = call_attribute(self, getattr, name);
+        }
```

Comment:
I don't understand this comment. Could you explain please?

Reply:
With `_PyObject_GenericGetAttrWithDict(self, name, NULL, /*suppress=*/1);`,

`_PyObject_GenericGetAttrWithDict` will caught all `AttributeError`, and do not raise `AttributeError` at the end.

Other kinds of exceptions are untouched.

If no exception occurred with `res == NULL`, it means we already handle that in `_PyObject_GenericGetAttrWithDict`. This is the only case we should call the `call_attribute(self, getattr, name);`

Reply:
I may change this comment to
```suggestion
        /* if res == NULL with no exception set, then it must be an
           AttributeError suppressed by us. */
        if (res == NULL && !PyErr_Occurred()) {
            res = call_attribute(self, getattr, name);
        }
```

Reply:
Ah yes that's better. Now it makes sense. thanks.